### PR TITLE
 #162163007 Users should be able to follow each other

### PR DESCRIPTION
**Pull request by Luleherll**
#### What does this PR do?
- Enable user to follow other users
- Enable user to unfollow other users
- Write tests for user-following endpoints

#### Description of the tasks to be completed?
- Update `Profile` model in `Profiles` app
- Update `views` file in `Profiles` app

#### How should this be manually tested?
- Pull this branch to your local workspace
- Run `git checkout ft-user-following-162163007` command in the terminal to switch to this feature's branch
- Create a postgres database called `ahdb` and make migrations by running:
        - `python manage.py makemigrations`
        - `python manage.py migrate` 
- Run the server using `python manage.py runserver`

##### *Don't forget: Append the endpoints to the server URL*

- Register 2 new users by making `POST` requests with the user's JSON data to `api/users/` endpoint using Swagger documentation on `/docs` or POSTMAN.
```
Data format
{
  "user": {
        "email": "lule@dev.com",
        "username": "lule",
        "password": "luleadmin",
  }
}
```
- Copy and paste the token of the user you want to follow the other in the headers (if using POSTMAN).
```
Key = "Authorization"
value = "Token<space><token.here>"
```
## Follow user
- Make a `POST` request to `api/profiles/<username>/follow. Replace `<username>` with the user whose token you have copied and you should see the `following` keyword equal to `True`.
```
{
  "profile": {
    "username": "lule",
    "bio": "",
    "image": null,
    "following": true
  }
}
```
## Unfollow user
- Make a `DELETE` request to `api/profiles/<username>/follow. Replace `<username>` with the user whose token you have copied and you should see the `following` keyword equal to `false`.
```
{
  "profile": {
    "username": "lule",
    "bio": "",
    "image": null,
    "following": false
  }
}
```
#### Relevant pivotal tracker stories
#24

### CODE REVIEW

File: authors/apps/tests/test_profile.py [hunk 1]
```diff
@@ -37,3 +37,53 @@ def test_update_user_profile(self):
...
+            reverse('user_follow', kwargs={'username': 'Joan'}))
+        self.assertIn(
+            'Profile does not exist. Cross-check the provided username.',
+            response.data['errors']['detail'])
+
+        response = self.client.delete(
+            reverse('user_follow', kwargs={'username': 'Joan'}))
+        self.assertIn(
+            'Profile does not exist. Cross-check the provided username.',
+            response.data['errors']['detail'])
```
Comment (katunold) [inline authors/apps/tests/test_profile.py:102]: Well done Lule, but according to pep8 you need one empty line just below the last line

**Review by katunold — APPROVED** [review]
Well done Lule

**Review by okellogabrielinnocent — APPROVED** [review]

File: authors/apps/tests/test_profile.py (hunk 1 above)
Comment (Luleherll) [reply authors/apps/tests/test_profile.py:102]: It is there in the code, Github just doesn't include it in `Reader mode`.
image

**Review by okellogabrielinnocent — COMMENTED** [review]
It's better not to upload migrations to git for this will create conflicts

File: authors/apps/article/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Generated by Django 2.1.3 on 2018-12-10 09:49
```
Comment (billkabanga) [inline authors/apps/article/migrations/0001_initial.py]: You can remove the migrations file from the codebase as they help us avoid merge conflicts.

File: authors/apps/profiles/models.py [hunk 3]
```diff
@@ -20,6 +21,21 @@ class Profile(models.Model):
     def __str__(self):
         return self.user.username
 
+    def follow(self, profile):
```
Comment (billkabanga) [inline authors/apps/profiles/models.py:25]: Please add comments to the methods.

**Review by billkabanga — APPROVED** [review]

**Pull request merged**